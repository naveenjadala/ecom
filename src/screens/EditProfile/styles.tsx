import {StyleSheet, useWindowDimensions} from 'react-native';
import {getFontSize} from '../../assets/fonts/FontSizes';
import {colors} from '../../assets/colors';

export const editProfileStyles = StyleSheet.create({
  scroll: {
    flex: 1,
    backgroundColor: colors.primary,
  },
  container: {
    paddingTop: '5%',
    marginBottom: '3%',
  },
  profileImage: {
    alignItems: 'center',
    padding: 30,
  },
  profileName: {
    textAlign: 'center',
    color: 'black',
    fontSize: getFontSize(20),
  },
  profileElements: {
    flexDirection: 'row',
  },
  profileEvents: {
    backgroundColor: colors.primary,
  },
});
