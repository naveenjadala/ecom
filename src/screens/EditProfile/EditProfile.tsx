import {
  StyleSheet,
  Text,
  TextInput,
  View,
  ScrollView,
  TouchableOpacity,
} from 'react-native';
import React, {useState} from 'react';
import ProfileImage from '../../components/ProfileImage/ProfileImage';
import {editProfileStyles} from './styles';
import PhotoPickerModal from '../../components/imageUploadComponent/PhotoPickerModal/PhotoPickerModal';
import {Camera, Gallery} from '../../utils/camera';
import DoubleTextInput from '../../components/customTextInput/doubleTextInput/customTextInput';
import SingleTextInput from '../../components/customTextInput/singleTextInput/SingleTextInput';
import EditProfileHeader from './Components/EditProfileHeader';
import {useNavigation} from '@react-navigation/native';
import {ProfileStackNavigationProp} from '../../navigation/Route';

type Props = {};

/**
 * EditProfile component renders the user's profile with image, name, and other events.
 *
 * @param {Props} props - the props object containing the necessary data
 * @return {JSX.Element} the rendered user profile component
 */
const EditProfile = (props: Props) => {
  const {navigate, goBack} = useNavigation<ProfileStackNavigationProp>();
  const [openModal, setOpenModal] = useState<boolean>(false);
  const [image, setImage] = useState<string | undefined>('');

  /**
   * Function to open and close the image modal.
   */
  const openImageModal = () => {
    setOpenModal(prevState => !prevState);
  };

  /**
   * Opens the camera and sets the image using the first asset.
   *
   * @return {Promise<void>}
   */
  const openCamera = async () => {
    const camera = await Camera();
    setImage(camera.assets?.[0].uri);
    setOpenModal(false);
  };

  /**
   * Opens the gallery, selects the first image, sets the image, and closes the modal.
   *
   * @return {Promise<void>}
   */
  const openGallery = async () => {
    const camera = await Gallery();
    setOpenModal(false);
    setImage(camera.assets?.[0].uri);
  };

  const onCancel = () => {
    goBack();
  };

  const onSave = () => {
    navigate('Profile');
  };

  return (
    <ScrollView style={editProfileStyles.scroll}>
      <>
        <EditProfileHeader onCancel={onCancel} onSave={onSave} />
        <View style={editProfileStyles.container}>
          <View style={editProfileStyles.profileImage}>
            <ProfileImage onClick={openImageModal} image={image} />
            <Text style={{color: 'black'}}>Edit</Text>
            <DoubleTextInput
              label="Name"
              placeholderOne="First Name"
              placeholderTwo="Last Name"
              containerStyle={{marginVertical: 10}}
            />
            <SingleTextInput
              placeholder="Town/City, Country/Region"
              label="Hometown"
              containerStyle={{marginVertical: 10}}
            />

            <SingleTextInput
              placeholder="150 characters"
              label="Bio"
              multiline
              inputStyle={{height: 100, textAlignVertical: 'top'}}
              containerStyle={{marginVertical: 10}}
            />
          </View>

          <PhotoPickerModal
            visible={openModal}
            closeModal={openImageModal}
            openCamera={openCamera}
            openGallery={openGallery}
          />
        </View>
      </>
    </ScrollView>
  );
};

export default EditProfile;

const styles = StyleSheet.create({});
