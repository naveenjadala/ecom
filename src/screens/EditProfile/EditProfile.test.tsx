import {useNavigation} from '@react-navigation/native';
import {
  fireEvent,
  render,
  screen,
  useNavigationMock,
  waitFor,
} from '../../test/test-utils';
import EditProfile from './EditProfile';

const mockFn = jest.fn();

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    navigate: mockFn,
    goBack: jest.fn(),
  }),
}));

describe.only('should render EditProfile', () => {
  test('should render EditProfile', () => {
    render(<EditProfile />);
    expect(screen).toMatchSnapshot();
  });

  test('should render Header element', () => {
    render(<EditProfile />);
    const getHeader = screen.getByTestId('editProfileHeader');
    expect(getHeader).toBeTruthy();
  });
});
