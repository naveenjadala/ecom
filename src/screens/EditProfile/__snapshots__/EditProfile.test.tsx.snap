// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditProfile should render EditProfile 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <RCTSafeAreaView
      style={
        {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "minHeight": "100%",
          "minWidth": "100%",
        }
      }
    >
      <RCTScrollView
        style={
          {
            "backgroundColor": "#FFFFFF",
            "flex": 1,
          }
        }
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "elevation": 2,
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
                "padding": 15,
              }
            }
            testID="editProfileHeader"
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
              testID="cancelButton"
            >
              <Text
                style={
                  {
                    "color": "#000000",
                    "fontSize": 8,
                  }
                }
              >
                Cancel
              </Text>
            </View>
            <Text
              onPress={[Function]}
              style={
                {
                  "color": "#000000",
                  "fontSize": 8,
                }
              }
            >
              Save
            </Text>
          </View>
          <View
            style={
              [
                {
                  "backgroundColor": "rgba(202, 196, 208, 1)",
                  "height": 0.5,
                },
                undefined,
                false,
                false,
                undefined,
              ]
            }
          />
          <View
            style={
              {
                "marginBottom": "3%",
                "paddingTop": "5%",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "padding": 30,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "borderRadius": 100,
                    "height": 100,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 100,
                  }
                }
                testID="profileImage"
              >
                <test-file-stub
                  height={100}
                  width={100}
                />
              </View>
              <Text
                style={
                  {
                    "color": "black",
                  }
                }
              >
                Edit
              </Text>
              <View
                style={
                  {
                    "flex": 1,
                    "marginVertical": 10,
                    "width": "100%",
                  }
                }
              >
                <Text>
                  Name
                </Text>
                <TextInput
                  containerStyle={
                    {
                      "marginVertical": 10,
                    }
                  }
                  label="Name"
                  placeholder="First Name"
                  placeholderOne="First Name"
                  placeholderTwo="Last Name"
                  style={
                    {
                      "borderColor": "black",
                      "borderLeftWidth": 0.5,
                      "borderRightWidth": 0.5,
                      "borderTopLeftRadius": 5,
                      "borderTopRightRadius": 5,
                      "borderTopWidth": 0.5,
                      "padding": 15,
                    }
                  }
                />
                <TextInput
                  containerStyle={
                    {
                      "marginVertical": 10,
                    }
                  }
                  label="Name"
                  placeholder="Last Name"
                  placeholderOne="First Name"
                  placeholderTwo="Last Name"
                  style={
                    {
                      "borderBottomLeftRadius": 5,
                      "borderBottomRightRadius": 5,
                      "borderColor": "black",
                      "borderWidth": 0.5,
                      "padding": 15,
                    }
                  }
                />
              </View>
              <View
                style={
                  {
                    "flex": 1,
                    "marginVertical": 10,
                    "width": "100%",
                  }
                }
              >
                <Text>
                  Hometown
                </Text>
                <TextInput
                  containerStyle={
                    {
                      "marginVertical": 10,
                    }
                  }
                  label="Hometown"
                  placeholder="Town/City, Country/Region"
                  style={
                    {
                      "borderColor": "black",
                      "borderRadius": 5,
                      "borderWidth": 0.5,
                      "padding": 15,
                    }
                  }
                />
              </View>
              <View
                style={
                  {
                    "flex": 1,
                    "marginVertical": 10,
                    "width": "100%",
                  }
                }
              >
                <Text>
                  Bio
                </Text>
                <TextInput
                  containerStyle={
                    {
                      "marginVertical": 10,
                    }
                  }
                  inputStyle={
                    {
                      "height": 100,
                      "textAlignVertical": "top",
                    }
                  }
                  label="Bio"
                  multiline={true}
                  placeholder="150 characters"
                  style={
                    {
                      "borderColor": "black",
                      "borderRadius": 5,
                      "borderWidth": 0.5,
                      "height": 100,
                      "padding": 15,
                      "textAlignVertical": "top",
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </RCTSafeAreaView>
  </View>
</RNCSafeAreaProvider>
`;
